1. Wczytaj dane z pliku heart.csv i zapoznaj się z nimi. Wygeneruj statystyki.

In [3]:
import pandas as pd

heart_data = pd.read_csv('heart.csv')

heart_data.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

2. Całkowicie losowo wygeneruj brakujące dane tak, aby w kolumnach age i chol brakowało po około 10% wartości.


3. Usuń rekordy zawierające jakiekolwiek brakujące wartości.

4. Dokonaj imputacji metodą średniej.

5. Porównaj otrzymane rezultaty z oryginalnymi danymi.


6. Dokonaj imputacji metodami uzupełniania braków poprzednią wartością, interpolacji, MICE i k-najbliższych sąsiadów.


7. Dla każdej z metod policz najmniejszy i największy błąd, a także miary MAE i RMSE.

8. Utwórz wykres pokazujący zależność między poziomem cholesterolu a wiekiem pacjenta dla danych oryginalnych i po przeprowadzonych imputacjach.

9. Przeprowadź analogiczne analizy dla 30% brakujących wartości.

10. Zbadaj hipotezę, że średni poziom cholesterolu wynosi 250.


11. Sprawdź, czy na podstawie poziomu cholesterolu jesteś w stanie rozpoznać płeć pacjenta.